pkg/resources/database/datasource_uri.go

```diff
@@ -252,22 +252,47 @@ func (d *DataSourceURI) Read(ctx context.Context, req datasource.ReadRequest, re
 
 	var uri string
 	var params map[string]interface{}
-
-	const adminUsername = "avnadmin"
+	var user string
 
 	switch data.Type.ValueString() {
 	case "kafka":
-		res, err := client.GetDBAASServiceKafka(ctx, data.Name.ValueString())
+		res, err := waitForDBAASService(
+			ctx,
+			client.GetDBAASServiceKafka,
+			data.Name.ValueString(),
+			func(s *exoscale.DBAASServiceKafka) string { return string(s.State) },
+		)
 		if err != nil {
 			resp.Diagnostics.AddError(
 				"Client Error",
-				fmt.Sprintf("Unable to read Database Service kafka: %s", err),
+				fmt.Sprintf("Unable to read Database Service Kafka: %s", err),
 			)
 			return
 		}
 
 		uri = res.URI
 		params = res.URIParams
+		if i, ok := params["user"]; ok {
+			if s, ok := i.(string); ok {
+				user = s
+			}
+		} else {
+			resp.Diagnostics.AddError(
+				"Client Error",
+				"Database Service Kafka parameter user is missing",
+			)
+			return
+		}
```

Comment:
Way you implemented it only non-existing 'user' key is caught, but not non-string value (due to placement if else clause).
```suggestion
		if i, ok := params["user"]; ok {
			if s, ok := i.(string); ok {
				user = s
			}
		}
		if user == "" {
			resp.Diagnostics.AddError(
				"Client Error",
				"Database Service Kafka parameter user is missing",
			)
			return
		}
```